addons/pos_loyalty/static/src/js/Loyalty.js

```diff
@@ -847,11 +847,13 @@ const PosLoyaltyOrder = (Order) => class PosLoyaltyOrder extends Order {
                                 || line.ignoreLoyaltyPoints({ program })) {
                                 continue;
                             }
-                            let price_to_use = line.get_price_with_tax();
+
+                            let pointsPerUnit = 0;
                             if (program.program_type === 'gift_card') {
-                                price_to_use = line.price;
+                                pointsPerUnit = line.price;
+                            } else {
+                                pointsPerUnit = round_precision(rule.reward_point_amount * line.get_price_with_tax() / line.get_quantity(), 0.01);
                             }
```

Comment:
I think this whole thing should just be:
```js
const pointsPerUnit = round_precision(rule.reward_point_amount * line.get_price_with_tax() / line.get_quantity(), 0.01);
```

I don't understand why 'gift_card' is special here.

Reply:
Gift cards value is based on points rewarded. So if you apply a discount (for example a 50$ gift card that only cost 20$) the value of the gift card would be the discounted value. So when we compute the points for a gift card we always take the base value of the product

Reply:
I asked DAMA and he said we should only look at the "total price" in the line and we base the calculation of the gift card there. Whether global discount or line discount is applied, just look at the total price of the gift card line and base the calculation of "loyalty-points-based-on-monetary" there.